### ISSUE TITLE
Using unicode characters results in str references containing invalid UTF-8 (UB)

### ISSUE THREAD

**Issue by Alxandr** [maciejhirsz/logos#138]
The following code should illustrate the issue:

```rust
#[test]
fn test() {
  use logos::Logos;

  #[derive(Logos, Debug, PartialEq)]
  enum Test {
    #[token("a")]
    A,

    #[error]
    Error,
  }

  let mut lex = Test::lexer("💩");
  let tok = lex.next();
  let bytes = lex.slice().as_bytes();
  println!("bytes: {:?}", bytes);

  // all lines bellow here throws
  let s = std::str::from_utf8(bytes).unwrap();
  assert_eq!(s, "💩");
  assert_eq!(lex.span(), 0..4);
}
```

Basically, trying to tokenize the string `"💩"` results in an `&str` reference being returned from the lexer which is not valid UTF-8. This is UB in rust and should really not happen.

**Comment by maciejhirsz** [comment]
Good catch! UTF-8 char bounds were already checked once lexer is bumped, [but I missed it on first byte](https://github.com/maciejhirsz/logos/pull/139/commits/dba01de29b7e104e91e686f245373bebfee4c4d2).

**Comment by maciejhirsz** [comment]
Released a fix in 0.11.3 (`logos-derive` 0.11.5)
